website_event_sale_check_order_before_payment/controllers/main.py

```diff
@@ -0,0 +1,85 @@
+import logging
+
+from odoo import _, http
+from odoo.http import request
+
+import odoo.addons.website_sale.controllers.main as website_sale
+
+_logger = logging.getLogger(__name__)
+
+
+class WebsiteSale(website_sale.WebsiteSale):
+    @http.route(
+        ["/shop/payment/order_validity"], type="json", auth="public", website=True
+    )
+    def check_order_validity(self):
+        order = request.website.sale_get_order()
+        if order:
+            if order.state != "draft":
```

Comment:
```suggestion
        if order and order.state != "draft":
```

Reply:
I'm not sure of it... A big part of the function is in the `if order` statement, and not in the `if order.state != "draft"` (line 24 to 64)